packages/components/src/focal-point-picker/types.ts

```diff
@@ -0,0 +1,84 @@
+/**
+ * External dependencies
+ */
+import type {
+	CSSProperties,
+	ReactEventHandler,
+	Ref,
+	VideoHTMLAttributes,
+} from 'react';
+
+/**
+ * Internal dependencies
+ */
+import type { BaseControlProps } from '../base-control/types';
+
+export type FocalPoint = Record< FocalPointAxis, number >;
+export type FocalPointAxis = 'x' | 'y';
+
+export type FocalPointPickerProps = Pick<
+	BaseControlProps,
+	'help' | 'hideLabelFromVision' | 'label'
+> & {
+	/**
+	 * Autoplays HTML5 video. This only applies to video sources (`url`).
+	 *
+	 * @default true
+	 */
+	autoPlay?: boolean;
+	/**
+	 * Callback which is called when the focal point changes.
+	 */
+	onChange: ( value: FocalPoint ) => void;
+	onDrag?: ( value: FocalPoint, event: MouseEvent ) => void;
+	/**
+	 * Callback which is called at the end of drag operations.
+	 */
+	onDragEnd?: () => void;
+	/**
+	 * Callback which is called at the start of drag operations.
+	 */
+	onDragStart?: (
+		value: FocalPoint,
+		event: React.MouseEvent< Element, MouseEvent >
+	) => void;
+	/**
+	 * Function which is called before internal updates to the value state.
+	 * It receives the upcoming value and may return a modified one.
+	 */
+	resolvePoint?: ( point: FocalPoint ) => FocalPoint | undefined;
+	/**
+	 * URL of the image or video to be displayed.
+	 */
+	url: string;
+	/**
+	 * The focal point. Should be an object containing `x` and `y` params.
+	 */
+	value: FocalPoint;
+};
+
+export type FocalPointPickerControlsProps = {
+	onChange?: ( value: FocalPoint ) => void;
+	point?: FocalPoint;
+};
+
+export type FocalPointPickerGridProps = {
+	bounds: { width: number; height: number };
+	showOverlay?: boolean;
+};
+
+export type FocalPointPickerMediaProps = Pick<
+	VideoHTMLAttributes< HTMLVideoElement >,
+	'autoPlay' | 'muted'
+> & {
+	alt: string;
+	mediaRef?: Ref< any >;
+	onLoad?: ReactEventHandler< any >;
```

Comment:
Should we be more detailed here, or is it going to hurt us in the future since we'd be kinda exposing implementation details?

```suggestion
	onLoad?: ReactEventHandler< HTMLVideoElement | HTMLImageElement >;
```

Reply:
Oh nice, I thought that would cause complications when passing `onLoad` to both `video` and `img` elements, but I guess it's fine? Interesting how that `onLoad` is ok but `mediaRef` errors when you try the same thing.

Reply:
Yeah that is more complicated because `mediaRef` is passed to a `div`, an `img` and a `video` tag

<details>

<summary>We could do something like this</summary>

```diff
diff --git a/packages/components/src/focal-point-picker/media.tsx b/packages/components/src/focal-point-picker/media.tsx
index 1138781672..55ec6b510f 100644
--- a/packages/components/src/focal-point-picker/media.tsx
+++ b/packages/components/src/focal-point-picker/media.tsx
@@ -1,3 +1,8 @@
+/**
+ * External dependencies
+ */
+import type { Ref } from 'react';
+
 /**
  * Internal dependencies
  */
@@ -20,7 +25,7 @@ export default function Media( {
 		return (
 			<MediaPlaceholder
 				className="components-focal-point-picker__media components-focal-point-picker__media--placeholder"
-				ref={ mediaRef }
+				ref={ mediaRef as Ref< HTMLDivElement > }
 				{ ...props }
 			/>
 		);
@@ -36,7 +41,7 @@ export default function Media( {
 			loop
 			muted={ muted }
 			onLoadedData={ onLoad }
-			ref={ mediaRef }
+			ref={ mediaRef as Ref< HTMLVideoElement > }
 			src={ src }
 		/>
 	) : (
@@ -45,7 +50,7 @@ export default function Media( {
 			alt={ alt }
 			className="components-focal-point-picker__media components-focal-point-picker__media--image"
 			onLoad={ onLoad }
-			ref={ mediaRef }
+			ref={ mediaRef as Ref< HTMLImageElement > }
 			src={ src }
 		/>
 	);
diff --git a/packages/components/src/focal-point-picker/types.ts b/packages/components/src/focal-point-picker/types.ts
index 9619f9a61c..f458c9e9a6 100644
--- a/packages/components/src/focal-point-picker/types.ts
+++ b/packages/components/src/focal-point-picker/types.ts
@@ -69,7 +69,10 @@ export type FocalPointPickerMediaProps = Pick<
 	'autoPlay' | 'muted'
 > & {
 	alt: string;
-	mediaRef?: Ref< any >;
+	mediaRef?:
+		| Ref< HTMLVideoElement >
+		| Ref< HTMLImageElement >
+		| Ref< HTMLElement >;
 	onLoad?: ReactEventHandler< HTMLVideoElement | HTMLImageElement >;
 	src: string;
 };
```
</details>